platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -0,0 +1,63 @@
+#import "NSPredicate+MGLAdditions.h"
+
+class FilterEvaluator {
+public:
+    
+    mbgl::Value getValue(id obj) {
```

Comment:
Instead of a catch-all `getValue()` method, use the Visitor pattern that we’re already inside. Define operators for NSString and NSNumber in a FilterExpressionEvaluator, then apply the FilterExpressionEvaluator to each `constantValue` you encounter.